src/basic/pager.c

```diff
@@ -41,10 +43,45 @@ _noreturn_ static void pager_fallback(void) {
         _exit(EXIT_SUCCESS);
 }
 
+static int no_quit_on_interrupt(int exe_name_fd, const char *less_opts) {
+        _cleanup_fclose_ FILE *file = NULL;
+        _cleanup_free_ char *line = NULL;
+        int r;
+
+        assert(exe_name_fd >= 0);
+        assert(less_opts);
+
+        /* This takes ownership of exe_name_fd */
+        file = fdopen(exe_name_fd, "r");
+        if (!file)
+                return log_debug_errno(errno, "Failed to create FILE object: %m");
```

Comment:
hmm, one more nitpick: this error path should close exe_name_fd, so that it isn't leak in case fdopen() fails.